src/components/pages/Cases/CaseTable.js

```diff
@@ -267,59 +275,56 @@ export default function CaseTable(props) {
       ),
     },
   ];
-  const findRowByID = (selectedRowID, caseData) => {
-    caseData.forEach(row => {
-      if (row.case_id === selectedRowID) {
-        return row;
-      } else {
-        return 'Please select row to save.';
+
+  const findRowByID = (rowID, rowData) => {
+    for (let i = 0; i < rowData.length; i++) {
+      let currentRow = rowData[i];
+
+      if (currentRow.case_id === rowID) {
+        return currentRow;
       }
-    });
+    }
+    return 'Please select case to save.';
   };
 
   const postBookmark = rowToPost => {
     axiosWithAuth()
       .post(`/profile/${userInfo.sub}/case/${rowToPost}`, rowToPost)
-      // .then(res => {
-      //   let justAdded = res.data.case_bookmarks.slice(-1); // response comes back as array of all existing bookmarks
-      //   let justAddedID = justAdded[0].case_id;
-      //   let wholeAddedRow = findRowByID(justAddedID, caseData);
-      //   setSavedCases([...savedCases, wholeAddedRow]);
-
       .then(res => {
-        console.log(res.data);
         setSavedCases(res.data.case_bookmarks);
       })
       .catch(err => {
         console.log(err);
       });
   };
 
-  const bookmarkCases = selectedRowID => {
-    // loop through currently selected cases and do post requests
-    // need to reference rows by id, as that is all that selection stores
-    // need to account for duplicates as well
+  const bookmarkCases = targetRows => {
     let bookmarks = [];
-    if (selectedRowID.length === 0) {
-      alert('Please select cases(s) to be saved');
+    if (targetRows.length === 0) {
+      alert('Please select case(s) to be saved');
     } else {
-      selectedRowID.forEach(row => bookmarks.push(findRowByID(row, caseData)));
-    }
+      for (let i = 0; i < targetRows.length; i++) {
+        bookmarks.push(findRowByID(targetRows[i], caseData));
+      }
 
-    let savedIds = [];
-    if (savedCases) {
-      savedCases.forEach(id => savedIds.push(id.case_id));
-    }
-    bookmarks.forEach(b => {
-      if (savedIds.includes(b.case_id)) {
-        console.log('Case already saved to bookmarks');
-      } else {
-        postBookmark(b);
+      let savedIds = [];
+      for (let i = 0; i < savedCases.length; i++) {
+        savedIds.push(savedCases[i].case_id);
       }
-    });
-    alert('Cases Successfully Saved');
+      for (let i = 0; i < bookmarks.length; i++) {
+        if (savedIds.includes(bookmarks[i].case_id)) {
+          console.log('Case already saved to bookmarks');
+          continue;
+        } else {
+          postBookmark(bookmarks[i]);
+        }
+      }
+      alert('Case(s) Successfully Saved');
+    }
```

Comment:
Where we differed here is that I changed some references in this portion of the code. I also kept your forEach method as they both work the same in this case but I thought the way you had it before was more readable!